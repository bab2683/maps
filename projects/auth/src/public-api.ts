/*
 * Public API Surface of auth
 */

export * from './lib/auth.model';
export * from './lib/auth.module';
export * from './lib/store';
