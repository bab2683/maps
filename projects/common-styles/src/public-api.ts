/*
 * Public API Surface of common-styles
 */
export * from './lib/common-styles.module';
