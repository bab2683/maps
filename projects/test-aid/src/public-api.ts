/*
 * Public API Surface of test-aid
 */

export * from './lib/angular-fire.mock';
export * from './lib/rxjs.mock';
export * from './lib/utils.mock';
