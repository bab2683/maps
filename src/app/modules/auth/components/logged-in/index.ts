export * from './logged-in.component';
