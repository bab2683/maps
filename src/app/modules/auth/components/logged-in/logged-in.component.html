<ng-container *ngIf="user">
  Welcome {{ user.email }} <button type="button" (click)="logout.emit()">Logout</button>
</ng-container>
