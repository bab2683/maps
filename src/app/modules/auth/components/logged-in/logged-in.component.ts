import { Component, Input, OnInit } from '@angular/core';
import { User } from 'firebase/app';

import { AuthService } from '../../auth.service';

@Component({
  selector: 'auth-logged-in',
  templateUrl: './logged-in.component.html',
  styleUrls: ['./logged-in.component.scss']
})
export class LoggedInComponent implements OnInit {
  @Input() user: User;

  constructor(private auth: AuthService) {}

  public ngOnInit(): void {
    console.log(this.user);
  }

  public logout(): void {
    this.auth.logout();
  }
}
