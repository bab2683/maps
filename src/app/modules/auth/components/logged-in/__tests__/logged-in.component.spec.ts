import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LoggedInComponent } from '../logged-in.component';

describe('LoggedInComponent', () => {
  let component: LoggedInComponent;
  let fixture: ComponentFixture<LoggedInComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LoggedInComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoggedInComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
