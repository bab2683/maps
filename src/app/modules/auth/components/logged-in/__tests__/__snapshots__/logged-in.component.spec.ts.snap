// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInComponent should match snapshot 1`] = `
<auth-logged-in
  logout={[Function EventEmitter]}
>
  
</auth-logged-in>
`;
